src/scene/globalScene.ts

```diff
@@ -8,15 +12,19 @@ export interface SceneStateCallbackRemover {
   (): void;
 }
 
-class SceneState {
+class GlobalScene {
   private callbacks: Set<SceneStateCallback> = new Set();
 
   constructor(private _elements: readonly ExcalidrawElement[] = []) {}
 
-  getAllElements() {
+  getElementsIncludingDeleted() {
     return this._elements;
   }
 
+  getElements(): readonly NonDeletedExcalidrawElement[] {
+    return getNonDeletedElements(this._elements);
+  }
+
   replaceAllElements(nextElements: readonly ExcalidrawElement[]) {
```

Comment:
Any TS/Type theory expert that can figure out how to enforce we pass elements that were not returned from `getElements()` getter which returns only non-deleted elements, while here we want *all* elements? I didn't want to require a nominal type here, but maybe this is the only way to achieve it?